### DISCUSSION TITLE
Welcome to dkron Discussions!

### DISCUSSION THREAD

**Discussion by vcastellm** [dkron-io/dkron#863] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nirbd** [comment]
Hi,
I have a few questions
I've set up Dkron in aws env using sudo dkron agent --disable-usage-stats --tag my_role=role1 --retry-join "provider=aws region=eu-west-1 access_key_id=XXX secret_access_key=YYY tag_key=dkron tag_value=join"

1. It seems to work but when the dkron server instance is terminated and a new one starts the jobs are lost. I thought the jobs data is kept in nodes so it should recover them?
If not - how can I save the jobs in db to make sure it continues them if the dkron node dies?
2. Where are the logs of cron results kept? Is it stored in local storage? I need to make sure it wont eat disk space and finally fill the instance storage. Is there some contron on how much data is kept ?
3. When using the config above (discovery based on aws tags) is it guaranteed to find the servers without having to manually set their IP like I saw in the clustering section https://dkron.io/usage/clustering/

Thanks !!

**Comment by myrfnrz** [comment]
Hi,
recently i found this dkron and is awesome and i have a question about the API of create or update job. how to write HTTP body on API to submit multiple job? for exampe i have {"name":"job1...} and  {"name":"job2...} at the same HTTP body.

thanks!!

**Reply by vcastellm** [reply]
Hi @myrfnrz, there is no support for that in the jobs endpoint, you can either create a bash script that creates several curl requests looping through the jobs, you can programmatically use any language client available to do something similar or you can try to craft a json file with the format accepted by the `/restore` endpoint https://dkron.io/usage/upgrade/

**Comment by bhushanibm** [comment]
Hi,

Can the DKRON server/agent work in kubernetes env ? If so, there might be certain times when the DKRON server/agent is down due to some maintenance and if jobs are scheduled at this point of time are going to be missed. When DKRON server/agent service comes back to running, will all the jobs which were missed original since the service were down, will it be re-triggerred ?

Thanks

**Reply by bhushanibm** [reply]
@vcastellm :- Can you reply to the thread above ?

**Reply by yvanoers** [reply]
Dkron can work in a kubernetes environment, if you're talking about putting it in some pods and letting it run from there.

Regardless of k8s, if at some point a schedule is missed because there was no Dkron leader (for whatever reason), the job will not be caught up. There is a feature request about this: #460 .

**Comment by bhushanibm** [comment]
Hi,

Is there a plan to extend DKRON to support NOSQL instead of storing the scheduling information in BuntDB(Disk) ?

Thanks

**Reply by bhushanibm** [reply]
@yvanoers  :- Can you reply to the thread above ?

**Reply by vcastellm** [reply]
Hi @bhushanibm, can you elaborate on this use case?

The internal Db of Dkron works really well and it's very tightly integrated.

**Comment by bhushanibm** [comment]
Hi,

Does the Kafka Executor has the following provision

a) To add Key/Value Pair to the Kafka Message Header
b) Does it integrate with AWS MSK Kafka through IAM Authentication

Thanks

**Reply by bhushanibm** [reply]
@yvanoers  :- Can you reply to the thread above ?

**Reply by vcastellm** [reply]
Hi @bhushanibm, Kafka executor is a community maintained plugin, and currently it doesn't support these features. Feel free to contribute to the project if you need the features. 

This is an example of a contribution to the Kafka executor https://github.com/distribworks/dkron/pull/1037/files
